src/github.com/getlantern/errors/errors.go

```diff
@@ -0,0 +1,641 @@
+/*
+Package errors defines error types used across Lantern project.
+
+  errors.ReportTo(myErrorReporter)
+  ...
+  if n, err := Foo(); err != nil {
+    errors.Wrap(err).Report()
+  }
+
+Wrap() method will try as much as possible to extract details from the error
+passed in, if it's errors defined in Go standard library. For application
+defined error type, at least the Go type name and what err.Error() returns
+will be recorded.
+
+Extra fields can be chained in any order, at any time.
+
+  func Foo() *Error {
+  	//...
+    return errors.New("some error").With("some_counter", 1).WithOp("connect")
```

Comment:
Looking at my previous comment, I think part of why the struct initializer looks more readable to me is because of the syntax highlighting and whitespace.